src/App.tsx

```diff
@@ -158,13 +158,36 @@ const App = () => {
 
 	const shadow = urlParams.get('shadow') !== 'false';
 
+	const form = { userCode: urlParams.get('user_code') || '' };
+
+	const [width, height] = [
+		['width', 'REACT_APP_FLOW_WIDTH'],
+		['height', 'REACT_APP_FLOW_HEIGHT']
+	].map(([key, envVar]) => {
+		const value = urlParams.get(key) ?? env[envVar];
+		let num = value !== undefined ? parseInt(value, 10) : undefined;
+
+		if (Number.isNaN(value)) num = undefined;
+
+		return num;
+	});
+
+	const hasWidthHeight = width !== undefined || height !== undefined;
+
 	const containerClasses = clsx({
 		'descope-base-container': shadow,
-		'descope-wide-container': isWideContainer,
-		'descope-login-container': !isWideContainer
+		'descope-wide-container': !hasWidthHeight && isWideContainer,
+		'descope-login-container': !hasWidthHeight && !isWideContainer
 	});
 
-	const form = { userCode: urlParams.get('user_code') || '' };
+	// See: https://web.dev/blog/viewport-units
+	// This is sensitive to mobile
+	const css: CSSProperties = {
+		width: width !== undefined ? `min(${width}px, 100dvw)` : undefined,
```

Comment:
who said the caller want to pass width is in px? why not letting them pass what they want as a string?